src/utils/cosmosdb_model_versioned.ts

```diff
@@ -75,13 +72,16 @@ export const incVersion = (version: NonNegativeInteger) =>
 export abstract class CosmosdbModelVersioned<
   T,
   TN extends Readonly<T & Partial<NewVersionedModel>>,
-  TR extends Readonly<T & RetrievedVersionedModel>
-> extends CosmosdbModel<T, TN & BaseModel, TR> {
+  TR extends Readonly<T & RetrievedVersionedModel>,
+  MODEL_ID_KEY extends keyof T,
```

Comment:
nit: it looks like by convention we are using upper cases types only for acronyms or single characters (ie. `T`, `TN`) while for extended names (ie. `BaseModel`) we're using PascalCase so i guess these may be better named as
`ModelIdKey` and `PartitionKey` (or `ModelIdKeyT`, `PartitionKeyT` in case of conflicts)

Reply:
correct, and it should have been obvious due the fact that we have
```ts
    protected readonly modelIdKey: ModelIdKey,
    protected readonly partitionKey?: PartitionKey
```